### PR TITLE
Properties of deeply inherited elements get lost

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses #542 and #590.

### CODE REVIEW

**Review by AlexandreSajus — APPROVED** [review]
Looks good to me

File: tools/_setup_generation/step_viselements.py [hunk 1]
```diff
@@ -86,8 +88,9 @@ def load_elements(self, elements_json_path: str, prefix: str, doc_pages_path: st
                                 default_property = property[__class__.NAME]
                             del property[__class__.DEFAULT_PROPERTY]
                 element_desc[__class__.DEFAULT_PROPERTY] = default_property
+
             # Resolve inheritance
-            def merge(element_desc, parent_element_desc, default_property) -> Optional[str]:
+            def merge(element_desc, parent_element_desc, default_property, FL_et, FL_pet) -> Optional[str]:
```
Comment (FredLL-Avaiga) [inline tools/_setup_generation/step_viselements.py]: FL & pet ?

File: tools/_setup_generation/step_viselements.py [hunk 2]
```diff
@@ -104,14 +107,21 @@ def merge(element_desc, parent_element_desc, default_property) -> Optional[str]:
                 if not default_property and parent_element_desc.get(__class__.DEFAULT_PROPERTY, False):
                     default_property = parent_element_desc[__class__.DEFAULT_PROPERTY]
                 return default_property
-            for element_type, element_desc in self.elements.items():
+            def resolve_inheritance(element_desc, FL_et):
```
Comment (FredLL-Avaiga) [inline tools/_setup_generation/step_viselements.py]: again

**Review by FredLL-Avaiga — APPROVED** [review]
🥇

File: tools/_setup_generation/step_viselements.py (hunk 1 above)
Comment (FabienLelaquais) [reply tools/_setup_generation/step_viselements.py]: Pfff. That was for debugging purpose. Thanks!!

File: tools/_setup_generation/step_viselements.py (hunk 2 above)
Comment (FabienLelaquais) [reply tools/_setup_generation/step_viselements.py]: Yep

**Pull request merged**